src/types/box.js

```diff
@@ -21,20 +23,31 @@ module.exports = function(Chart) {
 				right = chartArea.right, 
 				bottom = chartArea.bottom;
 
-			var min,max;
+			model.ranges = {};
+			var min, max;
 
 			if (xScale) {
-				min = isValid(options.xMin) ? xScale.getPixelForValue(options.xMin) : chartArea.left;
-				max = isValid(options.xMax) ? xScale.getPixelForValue(options.xMax) : chartArea.right;
+				min = helpers.isValid(options.xMin) ? xScale.getPixelForValue(options.xMin) : chartArea.left;
+				max = helpers.isValid(options.xMax) ? xScale.getPixelForValue(options.xMax) : chartArea.right;
 				left = Math.min(min, max);
 				right = Math.max(min, max);
+
+				model.ranges[options.xScaleID] = {
+					min: xScale.getValueForPixel(left),
```

Comment:
I was going to comment that you could use `xScale.min` and `xScale.max` but I realized I don't think we explicitly make those public so good call doing it this way